kore/src/Kore/Step/SMT/Representation/Symbols.hs

```diff
@@ -0,0 +1,214 @@
+{-|
+Module      : Kore.Step.SMT.Representation.Symbols
+Description : Builds an SMT representation for symbols.
+Copyright   : (c) Runtime Verification, 2019
+License     : NCSA
+Maintainer  : virgil.serbanuta@runtimeverification.com
+-}
+
+module Kore.Step.SMT.Representation.Symbols
+    ( buildRepresentations
+    ) where
+
+import qualified Data.Map as Map
+import           Data.Maybe
+                 ( mapMaybe )
+
+import           Kore.AST.Identifier
+                 ( Id )
+import           Kore.AST.Kore
+                 ( VerifiedKorePattern )
+import           Kore.AST.MetaOrObject
+                 ( Object )
+import           Kore.AST.Sentence
+                 ( SentenceSymbol (SentenceSymbol, sentenceSymbolResultSort, sentenceSymbolSorts, sentenceSymbolSymbol) )
+import qualified Kore.AST.Sentence as Sentence
+                 ( Symbol (Symbol) )
+import qualified Kore.AST.Sentence as Sentence.Symbol
+                 ( Symbol (..) )
+import qualified Kore.Attribute.Constructor as Attribute
+                 ( Constructor (Constructor, isConstructor) )
```

Comment:
Not that you need to change it, but I think that for fully-qualified imports, the explicit import lists are just noise.

Reply:
Sometimes yes, sometimes not. I don't have strong reasons for doing so, but there are a few advantages. A while ago we said that we'd import struct fields in a qualified import, preferably one with the struct name in it (just `field` would be wrong, and StructName.field is so much nicer than SomewhatRelatedPackageName.field). This means that:

a. There will be a qualified import just for `StructName(..)`.
b. Whenever we import something with a generic qualifier (e.g. `Attribute`), it makes some sense to have another import for the actual struct fields, with a specific qualifier (e.g. `Attribute.StructName`)
c. But then it makes sense to restrict the import with a generic qualifier (e.g. `SMT`) to only the things that we want to qualify with that (i.e. we would import `StructName(StructName)`, but we don't want to import all of `StructName`).
d. But, if I want to make this work, I need to import all things selectively, even when I'm using qualified imports.